core/src/main/java/org/elasticsearch/common/unit/Fuzziness.java

```diff
@@ -103,10 +114,28 @@ public static Fuzziness build(Object fuzziness) {
         String string = fuzziness.toString();
         if (AUTO.asString().equalsIgnoreCase(string)) {
             return AUTO;
+        }  else if (string.toUpperCase(Locale.ROOT).startsWith(AUTO.asString() + ":")) {
+            return parseCustomAuto(string);
         }
         return new Fuzziness(string);
     }
 
+    private static Fuzziness parseCustomAuto( final String string) {
+        String[] fuzzinessLimit = string.substring(AUTO.asString().length() + 1).split(",");
+        if (fuzzinessLimit.length == 2) {
+            try {
+                int lowerLimit = Integer.parseInt(fuzzinessLimit[0]);
+                int highLimit = Integer.parseInt(fuzzinessLimit[1]);
+                return new Fuzziness("AUTO", lowerLimit, highLimit);
+            } catch (NumberFormatException e) {
+                throw new ElasticsearchParseException("failed to parse [{}] as a \"auto:int,int\"", e,
+                    string);
+            }
+        } else {
+            throw new ElasticsearchParseException("Auto fuzziness wrongly configured");
```

Comment:
can you make the message more specific and say that you were looking for 2 integers, but got `Arrays.toString(fuzzinessLimit)`?